x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -158,7 +158,7 @@ private String getSessionIndex(Assertion assertion) {
 
     private void checkResponseDestination(Response response) {
         final String asc = getSpConfiguration().getAscUrl();
-        if (asc.equals(response.getDestination()) == false) {
+        if (response.isSigned() && asc.equals(response.getDestination()) == false) {
```

Comment:
I would prefer that we continue to reject _present-but-incorrect_ destinations even for unsigned messages.

So this should be more like
```
   if (asc.equals(response.getDestination()) == false) {
      if(response.isSigned() || Strings.hasText(response.getDestination()) {
         // ...
```